src/amo/components/UserProfileEdit/index.js

```diff
@@ -0,0 +1,475 @@
+/* @flow */
+import invariant from 'invariant';
+import * as React from 'react';
+import Textarea from 'react-textarea-autosize';
+import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import Link from 'amo/components/Link';
+import NotFound from 'amo/components/ErrorPage/NotFound';
+import {
+  editUserAccount,
+  fetchUserAccount,
+  getCurrentUser,
+  getUserByUsername,
+  hasPermission,
+} from 'amo/reducers/users';
+import { USERS_EDIT } from 'core/constants';
+import { withErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import { sanitizeHTML } from 'core/utils';
+import Button from 'ui/components/Button';
+import Card from 'ui/components/Card';
+import Notice from 'ui/components/Notice';
+import type { UsersStateType, UserType } from 'amo/reducers/users';
+import type { DispatchFunc } from 'core/types/redux';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { I18nType } from 'core/types/i18n';
+
+import './styles.scss';
+
+
+type Props = {|
+  currentUser: UserType | null,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  hasEditPermission: boolean,
+  i18n: I18nType,
+  isEditing: boolean,
```

Comment:
I find the name of this prop confusing. To me `isEditing` sounds like a user is looking at the edit screen and making changes, but really this means the user has submitted changes via the edit screen, correct? Maybe `isSaving` would be better?

Reply:
It would be indeed better but I did not change this variable name because it is bound to a property in the reducer, and I did not want to change that property. That would be too much changes. I decided not to just change this name here to keep the 1:1 relation.

Can we do this in another PR? I can open a new issue for that.

Reply:
Sure, that's fine. Please open an issue for that so the next person to review the code isn't also confused by it. ;-)

Reply:
Maybe it would be worth adding a TODO comment here as well, with a link to the issue.